controllers/csiaddonsnode_controller.go

```diff
@@ -58,5 +121,50 @@ func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 func (r *CSIAddonsNodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&csiaddonsv1alpha1.CSIAddonsNode{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// addFinalizer adds finalizer to csiAddonsNode if it is not present.
+func (r *CSIAddonsNodeReconciler) addFinalizer(
+	ctx context.Context,
+	logger *logr.Logger,
+	csiAddonsNode *csiaddonsv1alpha1.CSIAddonsNode) error {
+
+	if !util.ContainsInSlice(csiAddonsNode.Finalizers, csiAddonsNodeFinalizer) {
+		logger.Info("Adding finalizer")
+
+		csiAddonsNode.Finalizers = append(csiAddonsNode.Finalizers, csiAddonsNodeFinalizer)
+		if err := r.Client.Update(ctx, csiAddonsNode); err != nil {
+			logger.Error(err, "Failed to add finalizer")
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+// removeFinalizer removes finalizer from csiAddonsNode if it is present.
+func (r *CSIAddonsNodeReconciler) removeFinalizer(
+	ctx context.Context,
+	logger *logr.Logger,
+	csiAddonsNode *csiaddonsv1alpha1.CSIAddonsNode) error {
+
+	if util.ContainsInSlice(csiAddonsNode.Finalizers, csiAddonsNodeFinalizer) {
+		logger.Info("Removing finalizer")
+
+		csiAddonsNode.Finalizers = util.RemoveFromSlice(csiAddonsNode.Finalizers, csiAddonsNodeFinalizer)
+		if err := r.Client.Update(ctx, csiAddonsNode); err != nil {
+			logger.Error(err, "Failed to remove finalizer")
+
+			return err
+		}
+	}
+	return nil
+}
+
+// createKey returns "namespace/name" string to be used as key for connection pool.
+func (r *CSIAddonsNodeReconciler) createKey(csiAddonsNode *csiaddonsv1alpha1.CSIAddonsNode) string {
```

Comment:
this can be a function not a method as `CSIAddonsNodeReconciler` is not used.

Reply:
> this can be a function not a method as `CSIAddonsNodeReconciler` is not used.

removed as it was only used in one place